### PR TITLE
Cleanup transaction docs

### PR DESCRIPTION
**Pull request by tcharding**
Do some cleanups to the docs in `blockdata::transaction`. Patch 1 needs the most careful review please. The rest should not be too controversial.

### CODE REVIEW

File: src/blockdata/transaction.rs [hunk 1]
```diff
@@ -506,16 +497,16 @@ impl Transaction {
...
     #[cfg_attr(docsrs, doc(cfg(feature = "bitcoinconsensus")))]
     pub fn verify<S>(&self, spent: S) -> Result<(), script::Error>
         where S: FnMut(&OutPoint) -> Option<TxOut> {
         self.verify_with_flags(spent, ::bitcoinconsensus::VERIFY_ALL)
     }
 
-    /// Verify that this transaction is able to spend its inputs
-    /// The lambda spent should not return the same TxOut twice!
+    /// Verify that this transaction is able to spend its inputs.
+    /// The lambda spent should not return the same [`TxOut`] twice!
```
[the remark this answers is not in the document]
Comment (Kixunil) [reply src/blockdata/transaction.rs:509]: "closure" is more commonly used term and I think not using backticks is also confusing.

> The `spent` closure should not return the same [`TxOut`] twice!

Would be also useful to know the consequences of returning the same `TxOut` twice.

[the remark this answers is not in the document]
Comment (tcharding) [reply src/blockdata/transaction.rs:509]: oooh, thanks for the explanation. Will modify as suggested.

**Comment by tcharding** [comment]
Changes in force-push:
- Rebase on master
- Use comment text suggested by @Kixunil

**Comment by tcharding** [comment]
Changes in force-push
- Rebased on master
- Use `Self::foo` instead of `foo` in docs
- Conditionally import type for docs build

Epic fail, I did a docs PR and did not build the docs before pushing :(

**Review by apoelstra — APPROVED** [review]
ACK e503f1433104b6481ac686d40c0f874e555f3ade

**Review by dr-orlovsky — APPROVED** [review]
ACK e503f1433104b6481ac686d40c0f874e555f3ade

**Pull request merged**